### ISSUE TITLE
Rule suggestion: disallow use of void

### ISSUE THREAD

**Issue by nippur72** [palantir/tslint#2922]
For some reason TypeScript allows to use `void` as any other type, you can return it or pass it in a `any` function argument. For instance this is valid code even in `--strict` mode:
```ts
function a(): void { }
function b(): void {
   return a();  // this should not be allowed
}

console.log(b()); // this should not be allowed too
```
This is not very intuitive, I think people expect `void` to work like in C,C++ etc., so why not add a specific tslint rule? It should enforce that `void` is never used a value.

**Comment by ajafff** [comment]
https://palantir.github.io/tslint/rules/no-void-expression/https://palantir.github.io/tslint/rules/no-void-expression/

**Comment by nippur72** [comment]
Ops For some reason I thought that rule did something different.
